core/server/middleware/api/spam-prevention.js

```
@@ -1,129 +1,130 @@
-            rateSigninAttempts = config.rateSigninAttempts || 10;
-
-        if (req.body.username && req.body.grant_type === 'password') {
-            loginSecurity.push({ip: remoteAddress, time: currentTime, email: req.body.username});
-        } else if (req.body.grant_type === 'refresh_token' || req.body.grant_type === 'authorization_code') {
-            return next();
-        } else {
-            return next(new errors.BadRequestError({message: i18n.t('errors.middleware.spamprevention.noUsername')}));
-        }
-
-        // filter entries that are older than rateSigninPeriod
-        loginSecurity = _.filter(loginSecurity, function filter(logTime) {
-            return (logTime.time + rateSigninPeriod > currentTime);
-        });
-
-        // check number of tries per IP address
-        ipCount = _.chain(loginSecurity).countBy('ip').value();
-        deniedRateLimit = (ipCount[remoteAddress] > rateSigninAttempts);
+var ExpressBrute = require('express-brute'),
+    BruteKnex = require('brute-knex'),
+    knexInstance = require('../../data/db/connection'),
+    store = new BruteKnex({tablename: 'brute', createTable:false, knex: knexInstance}),
+    moment = require('moment'),
+    errors = require('../../errors'),
+    config = require('../../config'),
+    spam = config.get('spam') || {},
+    _ = require('lodash'),
+    spamPrivateBlog = spam.private_blog || {},
+    spamGlobalBlock = spam.global_block || {},
+    spamGlobalReset = spam.global_reset || {},
+    spamUserReset = spam.user_reset || {},
+    spamUserLogin = spam.user_login || {},
+
+    i18n = require('../../i18n'),
+    handleStoreError,
+    globalBlock,
+    globalReset,
+    privateBlog,
+    userLogin,
+    userReset,
+    logging = require('../../logging'),
+    spamConfigKeys = ['freeRetries', 'minWait', 'maxWait', 'lifetime'];
+
+handleStoreError = function handleStoreError() {
+    // TODO make a custom error?
+    return new errors.MaintenanceError({message: 'DB error'});
```

I wonder why you are using a "MaintenanceError" here.
How about "NoPermissionError"?

Plus: the original error is ignored at all. Do you think the original error is unimportant?If yes, that's fine 👍